### PR TITLE
FIX #182

### LINKED ISSUE #182: Follow up on pull request #177
Although I have already added my feedback as a comment in the merged pull request #177, I am not sure it is monitored anymore, and that's why I've created a new issue for this. 

I had the same underlying issue on many pages from Reuters, where the author in the schema was an object `"author":{"@type":"Person","byline":"Nia Williams"}`, which has no `name` key.

Although pull request #177 fixes the error that is raised and not caught, it sets the author name to "", which implies that it will ignore the author name from meta. `authors_from_schema` will not be empty, but "", so it will return that value, instead of using the one from `authors_from_meta`. In addition, it does not make sense to invoke `self.__get_authors_from_meta()` if `authors_from_meta` already has a value and takes precedence. A suggested code improvement is provided below.

```python
    def extract(self):
        authors_from_schema = self.__get_authors_from_schema()
        if authors_from_schema:
            return authors_from_schema
        return self.__get_authors_from_meta()
```

```python
    def __get_authors_from_schema(self):
        authors = list()
        if self.article.schema and "author" in self.article.schema:
            schema_authors = self.article.schema["author"]
            if isinstance(schema_authors, dict):
                schema_authors = [schema_authors]
            for author in schema_authors:
                if isinstance(author, dict):
                    author = author.get("name", None)
                    if author:
                        authors.append(author)
                else:
                    authors.append(author)
        return authors
```

### PR DESCRIPTION
**Pull request by erikvullings**
As requested, I''ve opened a PR to fix it.

### CODE REVIEW

File: goose3/extractors/authors.py [hunk 1]
```diff
@@ -70,7 +69,9 @@ def __get_authors_from_schema(self):
                 schema_authors = [schema_authors]
             for author in schema_authors:
                 if isinstance(author, dict):
-                    authors.append(author.get("name", ""))
+                    author = author.get("name", None)
+                    if author:
```
Comment (lababidi) [inline goose3/extractors/authors.py:73]: please use a different variable than author, it is the same as the loop variable `extracted_author` may be more appropriate

File: goose3/extractors/authors.py [hunk 2]
```diff
@@ -69,9 +69,9 @@ def __get_authors_from_schema(self):
                 schema_authors = [schema_authors]
             for author in schema_authors:
                 if isinstance(author, dict):
-                    author = author.get("name", None)
+                    extracted_author = author.get("name", None)
                     if author:
```
Comment (barrust) [inline goose3/extractors/authors.py:73]: Shouldn't this also be extracted author? Otherwise a `None` might be inserted into the array.

Comment (erikvullings) [reply goose3/extractors/authors.py]: You are completely right - stupid oversight on my part. Thanks for catching it!

**Comment by lababidi** [comment]
@barrust looks good. I would simplify eventually to `return from_schema() or from_meta()`

**Pull request merged**